### DISCUSSION TITLE
Blank DHCP Server Page

### DISCUSSION THREAD

**Discussion by GitHubUser655321** [RaspAP/raspap-webgui#1480] [General]
I have an issue where the DHCP Server page is blank.
When I installed RaspAP it worked fine, but after a while (not sure what I changed, if anything) the page would show but nothing would work on the page.
I updated to the latest version to see if that fixed anything and now the page is completely blank.

I've tried the uninstall script and then the quick install script again to see if it fixed anything, but didn't.

I don't see any errors in the web browser console and nothing in the debug file.

I have AdGuard Home and Emby installed too.

Here's some of the configs from debug log;

System Info
===========
Hardware: Raspberry Pi 4 Model B Rev 1.4
Detected OS: Raspbian GNU/Linux 11 (bullseye) 32-bit
Kernel: Linux RasPi 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux
System Uptime: 0 days, 0 hours, 27 minutes
Memory Usage: 15.4678%

Installed Packages
==================
PHP Version: 7.4.33 (cli) (built: Jun  9 2023 16:51:37) ( NTS )
Dnsmasq Version: 2.85
dhcpcd Version: 8.1.2
lighttpd Version: 1.4.59
vnStat Version: 2.6

RaspAP Install
==============
RaspAP Version: 3.0.4
RaspAP Installation Directory: /var/www/html
RaspAP hostapd.ini contents:
WifiInterface = wlan0
LogEnable = 0
WifiAPEnable = 0
BridgedEnable = 0
WifiManaged = wlan0
RaspAP provider.ini: Not present

dnsmasq status:
● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
     Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-12-15 09:46:03 MST; 23min ago
   Main PID: 2303 (dnsmasq)
      Tasks: 1 (limit: 1599)
        CPU: 263ms
     CGroup: /system.slice/dnsmasq.service
             └─2303 /usr/sbin/dnsmasq -x /run/dnsmasq/dnsmasq.pid -u dnsmasq -r /run/dnsmasq/resolv.conf -7 /etc/dnsmasq.d,.dpkg-dist,.dpkg-old,.dpkg-new --local-service --trust-anchor=.,20326,8,2,e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d

Dec 15 09:46:03 RasPi systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
Dec 15 09:46:03 RasPi systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.

dhcpcd status:
● dhcpcd.service - DHCP Client Daemon
     Loaded: loaded (/lib/systemd/system/dhcpcd.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-12-15 09:46:01 MST; 23min ago
       Docs: man:dhcpcd(8)
   Main PID: 2254 (dhcpcd)
      Tasks: 2 (limit: 1599)
        CPU: 1.287s
     CGroup: /system.slice/dhcpcd.service
             ├─2254 /usr/sbin/dhcpcd -b -q
             └─2269 wpa_supplicant -B -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0

**Comment by GitHubUser655321** [comment]
OK - so I figured out what/when this issue happens.
After I install AdGuard Home on the machine and then go to the 'DHCP Server Page > Advanced' tab and 'Add an upstream DNS server'.
Once I click the 'Save settings' button, the 'Only ever query DNS servers configured below' switch turns off the entry I added disappears and the 'Adapter IP Address Settings' changes to 'DHCP' and all the settings on the 'Server Settings' tab go blank.
Even if I re-enter all the info, it doesn't save or show the data.

**Reply by billz** [reply]
Sounds like a bug. I'm working on an exception handling class for RaspAP. Until then, try executing the following in the shell:

```
sudo tac /var/log/lighttpd/error.log | grep -m 1 'PHP Fatal error'
```

**Reply by GitHubUser655321** [reply]
That returned;

`(mod_fastcgi.c.449) FastCGI-stderr:PHP Fatal error:  Uncaught TypeError: explode(): Argument #2 ($string) must be of type string, array given in /var/www/html/ajax/networking/get_netcfg.php:14`

**Reply by billz** [reply]
I've captured this in a new issue. Thanks for reporting it!

**Reply by GitHubUser655321** [reply]
In the mean time, would it be possible for you to point me to the file(s) where the Upstream DNS Server gets add so I can add it manually for now?

**Reply by billz** [reply]
RaspAP writes upstream servers to the dnsmasq config for the active AP interface. So, assuming `wlan0` (the default) it would appear like so:

`/etc/dnsmasq.d/090_wlan0.conf`

```
# RaspAP wlan0 configuration
interface=wlan0
dhcp-range=10.3.141.50,10.3.141.254,255.255.255.0,12h
no-resolv
server=80.80.80.80
```
